Engine/media/audio/audio.cpp

```diff
@@ -39,8 +39,53 @@
 #include "main/game_run.h"
 
 using namespace AGS::Common;
+using namespace AGS::Engine;
+
+//-----------------------
+//sound channel management; all access goes through here, which can't be done without a lock
+
+static std::array<SOUNDCLIP *,MAX_SOUND_CHANNELS+1> _channels;
```

Comment:
I don't know if this gets null initialised or not for std::array.

Reply:
If I understood correctly from the answer: https://stackoverflow.com/questions/18295302/default-initialization-of-stdarray

We need to add empty initializer list `{}` to initialize all members to default, which is nullptr supposedly in this case.

Reply:
that topic doesnt address your question, which is affected by the fact that statics are zero-initialized. in this case std::array is probably built with a standard array internally allocated not-on-the-heap, and the standard array is zero-initialized due to the whole thing being static. 

As for whether you ought to initialize static variables, that's a matter of elegance and what type of aesthetics you like and what kind of confidence you want to instill on the reader.  To me, leaving off the initializer means 'danger! must think about what it means that there was no initializer was here'. In this case, the reader is supposed to know that statics are zero-initialized, but I can see why it might be doubtful in this case.

A rule where arrays should always be initialized would hamper performance. And if there's any case where variables can be confidently not initialized, it's global statics.

Reply:
Alright, got it.